skypy/pipeline/_config.py

```diff
@@ -30,7 +30,16 @@ def load(cls, stream):
         '''load the first YAML document from stream'''
         loader = cls(stream)
         try:
-            return loader.get_single_data()
+            single_data = loader.get_single_data()
+
+            keys = single_data.keys() if single_data else {}
+            for key in keys:
+                if not isinstance(key, str):
+                    error_message = f"Invalid key found in config.  {key} is not a string." \
+                                    f"  Either rename this value or wrap it in quotes."
+                    raise ImportError(error_message)
+
+            return single_data
```

Comment:
Thanks @JonathanDHarris looks good and I agree with the decision to check this in the conifg parser rather than the pipeline. Since the config can contain nested dictionaries of variable definitions we will need to check the keys recursively to ensure all parameters, tables, columns etc also have string names. And while we are at it we should also check that the kwargs for function calls also have string names. I've suggested a recursive implementation based off your code (untested) feel free to use or modify it. If anything is unclear just ask.

```suggestion
            def validate_keys(config):
                for k in config.keys():
                    if not isinstance(k, str):
                        raise ValueError(f"Invalid key found in config. {k} is not a string. "
                                         f"Either rename this value or wrap it in quotes.")
            
            from collections.abc import Mapping
            def validate_config(config):
                # Check each key at the current depth is a string
                validate_keys(config)
                for v in config.values():
                    # If any values are dictionaries, recurse
                    if isinstance(v, Mapping):
                        validate_config(v)
                    # If any values are tuples (i.e. function calls) validate kwargs
                    if isinstance(v, tuple) and len(v) > 1 and isinstance(v[1], Mapping):
                        validate_keys(v[1])
                return config

            return validate_config(single_data if single_data else {})
```